packages/react-core/src/components/Alert/AlertIcon.tsx

```diff
@@ -20,10 +21,12 @@ export interface AlertIconProps extends React.HTMLProps<HTMLDivElement> {
   variant: 'success' | 'danger' | 'warning' | 'info' | 'default';
   /** className */
   className?: string;
+  /** A custom icon. If not set the icon is set according to the variant */
+  customIcon?: React.ComponentClass<SVGIconProps>;
 }
 
-export const AlertIcon = ({ variant, className = '', ...props }: AlertIconProps) => {
-  const Icon = variantIcons[variant];
+export const AlertIcon = ({ variant, customIcon, className = '', ...props }: AlertIconProps) => {
+  const Icon = customIcon || variantIcons[variant];
   return (
     <div {...props} className={css(styles.alertIcon, className)}>
       <Icon />
```

Comment:
on second thought... we would be limiting the icon here to only allow react icons. how about we open this up to accept `children` so you're not limited:
```
<div {...props} className={css(styles.alertIcon, className)}>
  {children || <Icon />
</div>
```